src/main/java/io/vertx/core/impl/FutureImpl.java

```diff
@@ -78,29 +80,48 @@ public synchronized boolean isComplete() {
     return failed || succeeded;
   }
 
-  /**
-   * Set a handler for the result. It will get called when it's complete
-   */
-  public Future<T> setHandler(Handler<AsyncResult<T>> handler) {
-    boolean callHandler;
+  @Override
+  public Future<T> onComplete(Handler<AsyncResult<T>> h) {
+    if (h == null) {
```

Comment:
You might as well use https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-

Reply:
is requireNonNull intrisified ?

Reply:
https://stackoverflow.com/questions/31005009/objects-requirenonnullt-obj-instead-of-null-checks-and-manually-thrown-illegal#comment50039504_31005009

At the very least the the JIT will easily inline it.